scripts/build_powerplants.py

```diff
@@ -253,16 +253,41 @@ def add_custom_powerplants(ppl, inputs, config):
         return add_ppls
 
 
-def replace_natural_gas_technology(df):
-    mapping = {"Steam Turbine": "CCGT", "Combustion Engine": "OCGT"}
-    tech = df.Technology.replace(mapping).fillna("CCGT")
-    return df.Technology.mask(df.Fueltype == "Natural Gas", tech)
-
-
-def replace_natural_gas_fueltype(df):
-    return df.Fueltype.mask(
-        (df.Technology == "OCGT") | (df.Technology == "CCGT"), "Natural Gas"
+def replace_natural_gas_technology(df: pd.DataFrame):
+    """
+    Maps and replaces gas technologies in the powerplants.csv onto model
+    compliant carriers.
+    """
+    mapping = {
+        "Steam Turbine": "CCGT",
+        "Combustion Engine": "OCGT",
+        "NG": "CCGT",
+        "Ng": "CCGT",
+        "NG/FO": "OCGT",
+        "Ng/Fo": "OCGT",
+        "NG/D": "OCGT",
+        "LNG": "OCGT",
+        "CCGT/D": "CCGT",
+        "CCGT/FO": "CCGT",
+        "LCCGT": "CCGT",
+        "CCGT/Fo": "CCGT",
+    }
+    fueltype = df["Fueltype"] == "Natural Gas"
+    df.loc[fueltype, "Technology"] = (
+        df.loc[fueltype, "Technology"].replace(mapping).fillna("CCGT")
+    )
+    unique_tech_with_ng = df.loc[fueltype, "Technology"].unique()
+    unknown_techs = np.setdiff1d(unique_tech_with_ng, ["CCGT", "OCGT"])
+    if len(unknown_techs) > 0:
+        df.Technology.where(
+            df.Technology == "Natural Gas",
+            df["Technology"].map({t: "CCGT" for t in unknown_techs}),
+            inplace=True,
+        )
+    df["Fueltype"] = np.where(
+        df["Fueltype"] == "Natural Gas", df["Technology"], df["Fueltype"]
     )
```

Comment:
Cool :D
As this line however shall be removed as the fix is available in add_electricity and we may not wanted this here.
What do you think?

Reply:
Personally if it has to be one or the other I would think it's better to leave it here as it's clear the function is addressing natural gas fuel type and technology. The other bit of code is bundled with attach_conventional_carriers which thematically isn't connected to cleaning up the gas technology. Are there unintended consequences if we replace gas earlier on in the code?

Reply:
Ok to keep the fix here, however, we definitely need to remove the swap in the add_electricity as it would no more needed.
Moreover, to remind in add_electricity that that swap occurs, it would be appropriate to add a comment with a TODO and highlight that carrier/technology replacement for natural gas.

Similarly, a note shall be placed here as well because it is a hard-fix.

Reply:
@Tomkourou Hello, I agree that this conversation is resolved, but a commit is required to fix this.
Once that is done, we can merge it :)
If you could cover also the minor stuff below, that would be amazing :D

Reply:
done, I just forgot to push on the other stuff.